File: valid-parentheses/parkhojeong.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function (s) {
  console.log("test2")
  const pairs = { ")": "(", "]": "[", "}": "{" };
  const stack = [];

  for (const ch of s) {
    if (!pairs[ch]) {
      stack.push(ch);
      continue;
    }

    if (stack.pop() !== pairs[ch]) {
      return false;
    }
  }

  return stack.length === 0;
};

File: valid-parentheses/parkhojeong.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []

        close_parenthesis_map = {
            "[": "]",
            "(": ")",
            "{": "}"
        }

        for ch in s:
            if ch in close_parenthesis_map:
                stack.append(ch)
            else:
                if len(stack) == 0:
                    return False

                top = stack.pop()
                if close_parenthesis_map[top] != ch:
                    return False

        if len(stack) > 0:
            return False

        return True
